### ISSUE TITLE
Why sometimes AllenNLP does not log gradients information to tensorboard?

### ISSUE THREAD

**Issue by entslscheia** [allenai/allennlp#3870]
I first trained my model on a large dataset and found that the gradients of one layer were always 0. Then I tried to run experiments on a small dataset (only have 4 data points) to pinpoint the reason for it, however, I found that AllenNLP only logs gpu usage and some metric information without logging the gradients and parameter information, but I didn't change anything since the previous run. The only thing I changed is the path for training data. 
![image](https://user-images.githubusercontent.com/15921425/75509254-f6556b80-59b4-11ea-986a-8dff2510e72a.png)
Do you know what could be the possible reason for this?

**Comment by DeNeutoy** [comment]
You need to change this parameter:

https://github.com/allenai/allennlp/blob/master/allennlp/training/trainer.py#L146

We can't log every batch because the process is slow, so we only log every 100 batches by default.

**Comment by entslscheia** [comment]
> https://github.com/allenai/allennlp/blob/master/allennlp/training/trainer.py#L146

Thanks! This perfectly addresses my concern.
